### PR TITLE
[Alignment Configuration] Rename align_cfg::alignment_result_capture to align_cfg::detail::result_type 

### PR DESCRIPTION
**Pull request by simonsasse**
Resolves https://github.com/seqan/product_backlog/issues/183

This moves the alignment_result_capture to a new namespace seqan3::align_cfg::detail and then renames it to result_type. 
I needed to specify the whole namespace path seqan3::detail::XX for most of the view files. This is due to the newly created seqan3::align_cfg::detail which had a name clash with seqan3::detail.

### CODE REVIEW

File: include/seqan3/alignment/configuration/detail.hpp [hunk 1]
```diff
@@ -22,8 +22,8 @@ namespace seqan3::detail
  */
 enum struct align_config_id : uint8_t
 {
-    //!\brief ID for the \ref seqan3::align_cfg::alignment_result_capture "alignment_result_capture" option.
-    alignment_result_capture,
+    //!\brief ID for the \ref seqan3::align_config::detail::result_type "result_type" option.
```
Comment (simonsasse) [inline include/seqan3/alignment/configuration/detail.hpp:25]: Should be 
```suggestion
    //!\brief ID for the \ref seqan3::align_cfg::detail::result_type "result_type" option.
```

File: include/seqan3/alignment/configuration/align_config_result_type.hpp [hunk 2]
```diff
@@ -6,7 +6,7 @@
 // -----------------------------------------------------------------------------------------------------
 
 /*!\file
- * \brief Provides seqan3::align_cfg::alignment_result_capture.
+ * \brief Provides seqan3::detail::result_type.
```
Comment (simonsasse) [inline include/seqan3/alignment/configuration/align_config_result_type.hpp:9]: Should be:
```suggestion
 * \brief Provides seqan3::align_cfg::detail::result_type.
```

**Comment by simonsasse** [comment]
I already found two wrong lines and will change these when applying the first review.

**Comment by Irallia** [comment]
Because I'm rebasing on your branch (I got the same 'detail' namespace problems), I found some things to point out:
1. As far as I know, you should rebase this on `upstream/release-3.0.2` since this is part of the release.
2. You should add a changelog entry.
3. I think your change from `align_cfg` to `align_config` is not what was meant in the issue.

**Comment by simonsasse** [comment]
> Because I got the same 'detail' namespace problems, I found some things in your PR to point out:
> 
> 1. As far as I know, you should rebase this on `upstream/release-3.0.2` since this is part of the release.
Thanks, I'll do that ! 👍 
> 2. You should add a changelog entry.
Alright, that makes sense.
> 3. I think your change from `align_cfg` to `align_config` is not what was meant in the issue.
Thats not what I did. I moved 'align_cfg::alignment_result_capture' to 'align_cfg::detail:: alignment_result_capture' and renamed it to 'align_cfg::detail::result_type'. I thought this is what is wanted here?

File: include/seqan3/alignment/configuration/detail.hpp [hunk 3]
```diff
@@ -22,8 +22,8 @@ namespace seqan3::detail
...
+    result_type,
```
[the remark this answers is not in the document]
Comment (simonsasse) [reply include/seqan3/alignment/configuration/detail.hpp:26]: 1. Left it like this for now. If the design is gonna be changed, everything in this commit has to change anyways.
2. Changed it, but I also had to change the values in the compatibility table some lines below.

**Comment by Irallia** [comment]
> > 1. I think your change from `align_cfg` to `align_config` is not what was meant in the issue.
> 
> Thats not what I did. I moved 'align_cfg::alignment_result_capture' to 'align_cfg::detail:: alignment_result_capture' and renamed it to 'align_cfg::detail::result_type'. I thought this is what is wanted here?

Sry, I meant a single change, but I didn't find it again. I probably made a mistake.

File: include/seqan3/alignment/configuration/detail.hpp (hunk 1 above)
Comment (simonsasse) [reply include/seqan3/alignment/configuration/detail.hpp:25]: I think its supposed to be like this. In the belonging issue https://github.com/seqan/product_backlog/issues/183 it looks like this namespace `seqan3::align_cfg::detail` should be part of the public interface. I'll assign @rrahn as second reviewer, he wrote the issue.

**Comment by simonsasse** [comment]
no was my fault, I had a mistake somewhere here: https://github.com/seqan/seqan3/pull/2018#discussion_r467889757 :)

**Comment by simonsasse** [comment]
Due to a git mess @rrahn recommended to cherry pick the relevant commits and apply them to a fresh branch.

**Pull request closed without merging**